### PR TITLE
Add multiple jobs example

### PR DESCRIPTION
**Pull request by LHBruneton-C2C**
This PR adds support for multiple jobs monitoring, and illustrate its use through an example.

image

### CODE REVIEW

File: demo/examples/04-jobs.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      bars.jobsStatus = jobs;
+    })
+  )
+  .subscribe();
+
+btn.addEventListener('click', async () => {
+  // create a job, get a promise that resolves with the job id
+  const jobId = await queuePrint(spec.value);
+
+  getJobStatus(jobId).subscribe((printStatus) => {
```
Comment (fgravin) [inline demo/examples/04-jobs.js:26]: Is the Observable complete or do we need to unsubscribe somehow ?

File: demo/examples/04-jobs.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const bars = /** @type {CustomProgresses} */ root.querySelector(
+  'custom-progresses'
+);
+const spec = /** @type {PrintSpec} */ root.querySelector('print-spec');
+
+// subscribe to the long-running observable
+// beware of managing the unsubscription
+getJobsStatus()
+  .pipe(
+    tap((jobs) => {
```
Comment (fgravin) [inline demo/examples/04-jobs.js:16]: Don't know if it wouldn't be clearer to have it in the `subscribe` method ?

**Review by fgravin — APPROVED** [review]
Very good example thanks.
When the last job is finished, it keeps being in the queue, is there any issue with an Observable emission or is it execpted ?

**Comment by LHBruneton-C2C** [comment]
> When the last job is finished, it keeps being in the queue, is there any issue with an Observable emission or is it execpted ?

This is somehow due to the pairwise operator. We "lose" the first job status event, then compare if the job was completed in the previous event to filter it in tte current event. But when the last event is fired, there is no more event to filter the last completed job.
If you have an idea on how to solve this, I'd be glad to implement it.

File: demo/examples/04-jobs.js (hunk 1 above)
Comment (LHBruneton-C2C) [reply demo/examples/04-jobs.js:26]: `getJobStatus` (single job) does complete eventually, same as the other examples.

File: demo/examples/04-jobs.js (hunk 2 above)
Comment (LHBruneton-C2C) [reply demo/examples/04-jobs.js:16]: Yes, I took the habit to always do the maximum in the pipe operator, but here it might be too much.

**Pull request merged**